### PR TITLE
Fixes #7624: Technique copyGitFile can send success AND error reports on post-hook execution and lead to unexpected reports

### PR DESCRIPTION
**Pull request by jooooooon**
https://www.rudder-project.org/redmine/issues/7624

### CODE REVIEW

**Comment by jooooooon** [comment]
DO NOT MERGE

...yet. Needs testing and adapting to the other technique versions.

**Comment by jooooooon** [comment]
PR updated

File: techniques/fileDistribution/copyGitFile/1.3/copyFileFromSharedFolder.st [hunk 1]
```diff
@@ -110,13 +110,13 @@ bundle agent download_from_shared_folder
         ifvarclass => "!execute_command_${index}";
 
       "@@copyFile@@result_success@@${copyfile[${index}][uuid]}@@Post-modification hook@@${copyfile[${index}][name]}@@${g.execRun}##${g.uuid}@#${copyfile[${index}][destination]} was already in the desired state, so no command was executed"
-        ifvarclass => "execute_command_${index}.copy_file_${index}_kept.!copy_file_${index}_modified";
+        ifvarclass => "execute_command_${index}.!copy_file_${index}_failed.!copy_file_${index}_modified.copy_file_${index}_kept";
```
Comment (jooooooon) [inline techniques/fileDistribution/copyGitFile/1.3/copyFileFromSharedFolder.st]: This is the main bug. The success report could be sent even when the "failed" class was present, which of course made no sense.

File: techniques/fileDistribution/copyGitFile/1.3/copyFileFromSharedFolder.st [hunk 2]
```diff
@@ -110,13 +110,13 @@ bundle agent download_from_shared_folder
...
 
       "@@copyFile@@result_success@@${copyfile[${index}][uuid]}@@Post-modification hook@@${copyfile[${index}][name]}@@${g.execRun}##${g.uuid}@#The post-hook command for ${copyfile[${index}][destination]} was correctly executed"
-        ifvarclass => "copyfile_posthook_${index}_command_run_ok";
+        ifvarclass => "execute_command_${index}.copyfile_posthook_${index}_command_run_ok";
 
       "@@copyFile@@result_error@@${copyfile[${index}][uuid]}@@Post-modification hook@@${copyfile[${index}][name]}@@${g.execRun}##${g.uuid}@#The post-hook command for ${copyfile[${index}][destination]} couldn't be executed"
-        ifvarclass => "copyfile_posthook_${index}_command_run_failed";
+        ifvarclass => "execute_command_${index}.copyfile_posthook_${index}_command_run_failed";
```
Comment (jooooooon) [inline techniques/fileDistribution/copyGitFile/1.3/copyFileFromSharedFolder.st]: These changes are not strictly necessary but complete the combinatory if/not logic of the classes in this set of reports, making them a LOT easier to understand.

**Comment by jooooooon** [comment]
PR updated

**Comment by jooooooon** [comment]
This is ready for review now, thanks.

**Comment by peckpeck** [comment]
What about the 2.0 version in master ?

File: techniques/fileDistribution/copyGitFile/1.3/copyFileFromSharedFolder.st (hunk 1 above)
Comment (ncharles) [inline techniques/fileDistribution/copyGitFile/1.3/copyFileFromSharedFolder.st]: Do I understand correctly that you can have kept + failed at the same time ?
(reading code)
Oh, ok, permissions kept, copy failed - make sense

**Comment by ncharles** [comment]
This look ok for me.
The version 2.0 will need to be adressed as well inupper branch

**Comment by jooooooon** [comment]
> What about the 2.0 version in master ?

Huh, good point. How weird that we have 2.1 in 2.11, but no 2.0... I've opened https://www.rudder-project.org/redmine/issues/7643 to do this.

**Comment by jooooooon** [comment]
Self merging thanks to previous reviews.

**Pull request merged**